ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/PlanNode.java

```diff
@@ -192,10 +195,13 @@ static void throwKeysNotIncludedError(
       final LogicalSchema schema
```

Comment:
To better understand the PR. What is `columns` and what is `schema` exactly? Is `columns` the list of selected columns from the `SELECT` clause (in the corresponding order) ? Is `schema` the "physical" layout of the input (what is `schema` for a join query?)

Reply:
honestly, this code is super weird to me; I was thinking the same thing, but erred against changing it. It's only ever called like this: `orderColumns(getSchema().value(), getSchema());` so really, it could be done by just passing in one  schema and then ensuring it's properly ordered.